test/unit/iteration_test.rb

```diff
@@ -199,41 +156,169 @@ def foo
+    with_global_enforce_serializable_cursors(false) do
+      parent = build_invalid_cursor_job(cursor: UnserializableCursor.new)
+      parent.job_iteration_enforce_serializable_cursors = true
+      child = Class.new(parent)
+
+      assert_raises(ActiveJob::SerializationError) do
+        child.perform_now
+      end
+    end
+  end
+
+  def test_jobs_using_unserializable_cursor_will_not_raise_if_enforce_serializable_cursors_unset_per_class
+    with_global_enforce_serializable_cursors(true) do
+      job_class = build_invalid_cursor_job(cursor: UnserializableCursor.new)
+      job_class.job_iteration_enforce_serializable_cursors = false
+
+      assert_cursor_deprecation_warning_on_perform(job_class)
+    end
+  end
+
+  def test_jobs_using_unserializable_cursor_when_interrupted_should_only_store_the_cursor_position_and_no_serialized_cursor_position
+    job_class = build_invalid_cursor_job(cursor: UnserializableCursor.new)
+    with_interruption do
+      assert_cursor_deprecation_warning_on_perform(job_class)
+    end
+
+    job_data = ActiveJob::Base.queue_adapter.enqueued_jobs.last
+    refute_nil(job_data, "interrupted job expected in queue")
+    assert_instance_of(UnserializableCursor, job_data.fetch("cursor_position"))
+    refute_includes(job_data, "serialized_cursor_position")
+  end
+
+  def test_jobs_using_serializable_cursor_when_interrupted_should_store_both_legacy_cursor_position_and_serialized_cursor_position
+    job_class = build_invalid_cursor_job(cursor: SerializableCursor.new)
+    with_interruption do
+      assert_no_cursor_deprecation_warning_on_perform(job_class)
+    end
+
+    job_data = ActiveJob::Base.queue_adapter.enqueued_jobs.last
+    refute_nil(job_data, "interrupted job expected in queue")
+    assert_instance_of(SerializableCursor, job_data.fetch("cursor_position"))
+    assert_equal(ActiveJob::Arguments.serialize([SerializableCursor.new]).first,
+      job_data.fetch("serialized_cursor_position"))
+  end
+
+  def test_job_interrupted_with_only_cursor_position_should_resume
+    with_interruption do
+      InfiniteCursorLoggingJob.perform_now
+
+      work_one_job do |job_data|
+        job_data["cursor_position"] = "raw cursor"
+        job_data.delete("serialized_cursor_position")
+      end
+
+      assert_equal("raw cursor", InfiniteCursorLoggingJob.cursors.last)
+    end
+  ensure
+    InfiniteCursorLoggingJob.cursors.clear
+  end
+
+  def test_job_interrupted_with_serialized_cursor_position_should_ignore_unserialized_cursor_position
+    with_interruption do
+      InfiniteCursorLoggingJob.perform_now
+
+      work_one_job do |job_data|
+        job_data["cursor_position"] = "should be ignored"
+        job_data["serialized_cursor_position"] = ActiveJob::Arguments.serialize([SerializableCursor.new]).first
+      end
+
+      assert_instance_of(SerializableCursor, InfiniteCursorLoggingJob.cursors.last)
+    end
+  ensure
+    InfiniteCursorLoggingJob.cursors.clear
+  end
```

Comment:
I'm not sure if I'm missing something, but I wasn't able to find any tests that actually exercise resuming jobs. Everything seems to be testing running or interrupting jobs, but no tests actually resume anything. So this happens to add test coverage for that 😅